### PR TITLE
treasury info card

### PR DESCRIPTION
**Pull request by herbig**
## Description

Adds total USD to the DAO homepage

## Issue / Notion doc (if applicable)

closes https://github.com/decent-dao/fractal-interface/issues/661

## Testing

View a DAO with Safe assets that are worth USD.

## Screenshots (if applicable)

Screen Shot 2022-11-14 at 4 31 20 PM

### CODE REVIEW

File: src/pages/Treasury/hooks/useTreasuryTotalUSD.tsx [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { useFractal } from '../../../providers/fractal/hooks/useFractal';
+import { formatUSD } from '../../../utils/numberFormats';
+
+export function useTreasuryTotalUSD(): string {
+  const {
+    treasury: { assetsFungible },
+  } = useFractal();
+  let totalUSD = 0;
+  assetsFungible.map(asset => (totalUSD += Number(asset.fiatBalance)));
```
[the remark this answers is not in the document]
Comment (herbig) [reply src/pages/Treasury/hooks/useTreasuryTotalUSD.tsx]: Yeah good call. forEach makes a lot more sense

For reduce, it seemed like I need to return the same value at each iteration, which doesn't work with iterating through a list of `GnosisAssetFungible`

File: src/pages/DaoDashboard/Info/InfoTreasury.tsx [hunk 2]
```diff
@@ -28,7 +28,7 @@ export function InfoTreasury({}: IDAOGovernance) {
         textStyle="text-lg-mono-semibold"
         color="grayscale.100"
       >
-        {MOCK_TREASURY_TITLE}
+        {useTreasuryTotalUSD()}
```
Comment (Da-Colon) [inline src/pages/DaoDashboard/Info/InfoTreasury.tsx:31]: hooks shouldn't be used in the return here. the hook should return a variable that can be used here. 

```ts
const totalUSD = useTreasuryTotalUSD()

return (
   ...
    <Flex
	@@ -28,7 +28,7 @@ export function InfoTreasury({}: IDAOGovernance) {
        textStyle="text-lg-mono-semibold"
        color="grayscale.100"
      >
      {totalUSD}
      ...
```

Comment (Da-Colon) [reply src/pages/DaoDashboard/Info/InfoTreasury.tsx:31]: https://reactjs.org/docs/hooks-custom.html

Comment (Da-Colon) [reply src/pages/DaoDashboard/Info/InfoTreasury.tsx:31]: basically when you call a function in the render like this, everytime the component 're-renders' this method is called again and calculated each time. if you set the it to state or `useMemo` it won't be recalculated unless something has changed.

Comment (herbig) [reply src/pages/DaoDashboard/Info/InfoTreasury.tsx]: @Da-Colon ah, that makes sense, thanks

Does the latest update work?

**Review by Da-Colon — APPROVED** [review]

File: src/pages/DaoDashboard/Info/InfoTreasury.tsx (hunk 2 above)
Comment (Da-Colon) [reply src/pages/DaoDashboard/Info/InfoTreasury.tsx]: Works for me, sir 🚀

**Pull request merged**